File: mib_v2_3_2/Var.cpp
#include "Var.h"
#include <iostream>
#include <vector>
#include <set>
#include <optional>

using namespace std;

// Método para obtener los valores de la variable.
vector<int> Var::getValues() const {
    vector<int> result;
    if (event.has_value()) {
        result.push_back(event.value());
    } else {
        result.insert(result.end(), values.begin(), values.end());
    }
    return result;
}

// Método para resetear el evento.
void Var::reset() {
    event.reset();  
}

// Método opcional si se quiere establecer el evento manualmente.
void Var::setEvent(int newEvent) {
    event = newEvent;
}

// Método para obtener el valor del evento
int Var::getEvent() {
    return event.value();
}

// Método para obtener el nombre de la variable.
string Var::getName() const {
    return name;
}

File: mib_v2_3_3/var.py
class Var:
    """Clase para el manejo de las variables (establecer y manejar los eventos).
    Atributos:
        name (any): Nombre de la varibale.
        values (set): Conjunto, de enteros, que contiene los valores que representan un evento.
        event (any): Valor que representa un evento.   
        
    Var(any,set) -> Nuevo objeto Var.
    """
    def __init__(self,name,values:set):
        self.name = name
        self.values = values
        self.event = None
    
    def getValues(self) -> list:
        """ Método obtner los valores de la variable para calcular la marginal.

        Return:
            list: lista con los valores de la variable.
        """  
        if self.event:
            return [self.event]
        else:
            return list(self.values)
    
    def getCard(self) -> int:
        if self.event:
            return 1
        else:
            return len(self.values)
    
    def reset(self) -> None:
        self.event = None
    